### PR TITLE
 [Agent] Support hystrix plugin

### PR DESCRIPTION
**Pull request by ascrutae**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#612

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/HystrixCommandConstructorInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if (HystrixObservableCollapser.class.isAssignableFrom(objInst.getClass())) {
+            HystrixObservableCollapser hystrixCommand = (HystrixObservableCollapser)objInst;
+            identifyKey = hystrixCommand.getCollapserKey().name();
+        }
+
+        if (HystrixObservableCommand.class.isAssignableFrom(objInst.getClass())) {
+            HystrixObservableCommand hystrixCommand = (HystrixObservableCommand)objInst;
+            identifyKey = hystrixCommand.getCommandKey().name();
+        }
+
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/HystrixCommandConstructorInterceptor.java]: Is there a way to reorganize these `if`s ? Put them in a `if/else` way for improving performance?

File: apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/SWExecutionHookWrapper.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        EnhancedInstance enhancedInstance = (EnhancedInstance)commandInstance;
+        EnhanceRequireObjectCache enhanceRequireObjectCache = (EnhanceRequireObjectCache)enhancedInstance.getSkyWalkingDynamicField();
+        enhanceRequireObjectCache.setContextSnapshot(ContextManager.capture());
+        actual.onStart(commandInstance);
+    }
+
+    /**
+     * execution method
+     */
+
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/SWExecutionHookWrapper.java]: Empty line.

File: apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/SWExecutionHookWrapper.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    @Override public <T> void onExecutionSuccess(HystrixInvokable<T> commandInstance) {
+        ContextManager.stopSpan();
+        actual.onExecutionSuccess(commandInstance);
+    }
+
+    /**
+     * Fallback
+     */
+
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/SWExecutionHookWrapper.java]: empty line again.

File: apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/HystrixCommandConstructorInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        objInst.setSkyWalkingDynamicField(new EnhanceRequireObjectCache("Hystrix/" + identifyKey));
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/HystrixCommandConstructorInterceptor.java]: What is the `identifyKey` here? Can you give me some examples about your operation names?

**Comment by wu-sheng** [comment]
Review is done. Another requirement is your test report.

**Comment by ascrutae** [comment]
Here is the test report: https://github.com/SkywalkingTest/agent-integration-test-report/blob/master/2018/2/ascrutae/testReport-master-2018-02-27-11-12.md

**Comment by ascrutae** [comment]
Link Relate PR: #834

**Pull request merged**